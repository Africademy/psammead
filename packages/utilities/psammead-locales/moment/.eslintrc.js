module.exports = {
  rules: {
    // IE11 gives syntax error for trailing commas on functions and Moment
    // locales are not transpiled
    'comma-dangle': ['error', {
      'functions': 'never',
      'imports': 'only-multiline',
      'objects': 'only-multiline',
      'arrays': 'only-multiline',
    }],
    'object-shorthand': 0,
    'no-var': 0,
    'one-var': 0,
    'func-names': 0,
  },
};
