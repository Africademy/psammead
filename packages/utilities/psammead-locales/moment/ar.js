const moment = require('moment');
require('moment/locale/ar');

// the months have been overridden to maintain arabic dual month names
// other values in this locale have been left as set upstream
// see https://github.com/moment/moment/blob/develop/src/locale/ar.js

const dualMonthNames = [
  'يناير/ كانون الثاني',
  'فبراير/ شباط',
  'مارس/ آذار',
  'أبريل/ نيسان',
  'مايو/ أيار',
  'يونيو/ حزيران',
  'يوليو/ تموز',
  'أغسطس/ آب',
  'سبتمبر/ أيلول',
  'أكتوبر/ تشرين الأول',
  'نوفمبر/ تشرين الثاني',
  'ديسمبر/ كانون الأول',
];

moment.updateLocale('ar', {
  months: dualMonthNames,
  monthsShort: dualMonthNames,
  postformat(string) {
    return string.replace(/,/g, '،');
  },
});
