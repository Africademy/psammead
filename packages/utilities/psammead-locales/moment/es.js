var moment = require('moment');
require('moment/locale/es');

moment.updateLocale('es', {
  relativeTime: {
    past: '%s',
    m: '1 minuto',
    mm: '%d minutos',
    h: '1 hora',
    hh: '%d horas',
  },
});
