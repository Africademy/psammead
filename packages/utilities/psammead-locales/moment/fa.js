/* eslint-disable func-names */
const moment = require('moment');
const jalaaliHelper = require('./helpers/jalaali');

const symbolMap = {
  '1': '۱',
  '2': '۲',
  '3': '۳',
  '4': '٤',
  '5': '۵',
  '6': '۶',
  '7': '۷',
  '8': '۸',
  '9': '۹',
  '0': '۰',
};

const persianMonths = [
  'فروردین',
  'اردیبهشت',
  'خرداد',
  'تیر',
  'مرداد',
  'شهریور',
  'مهر',
  'آبان',
  'آذر',
  'دی',
  'بهمن',
  'اسفند',
];

const jalaliFormats = [
  'HH:mm', // LT FORMAT
  'HH:mm:ss', // LTS FORMAT
  'DD/MM/YYYY', // L FROMAT
  'D MMMM YYYY', // LL FROMAT
  'D MMMM YYYY HH:mm', // LLL FROMAT
  'dddd, D MMMM YYYY HH:mm', // LLLL FROMAT
];

moment().locale('fa');
moment.updateLocale('fa', {
  postformat(string) {
    const str = jalaaliHelper.addJalaliDate(
      'fa',
      persianMonths,
      jalaliFormats,
      string,
    );
    if (!str.includes('datetime =')) {
      return str
        .replace(/\d/g, function(match) {
          return symbolMap[match];
        })
        .replace(/,/g, '،');
    }
    return str;
  },
});
