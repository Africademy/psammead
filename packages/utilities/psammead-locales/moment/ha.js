var moment = require('moment');

var months = [
  'Janairu',
  'Fabrairu',
  'Maris',
  'Aprilu',
  'Mayu',
  'Yuni',
  'Yuli',
  'Agusta',
  'Satumba',
  'Oktoba',
  'Nuwamba',
  'Disamba',
];

moment.defineLocale('ha', {
  months: months,
  longDateFormat: { LL: 'D MMMM YYYY' },
  relativeTime: {
    past: '%s wuce',
    m: 'Minti 1 da ta',
    mm: 'Mintuna %d da suka',
    h: "Sa'a 1 da ta",
    hh: "Sa'o'i %d da suka",
  },
});
