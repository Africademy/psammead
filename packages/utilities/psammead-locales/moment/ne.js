var moment = require('moment');
require('moment/locale/ne');

moment.updateLocale('ne', {
  postformat: null,
  relativeTime: {
    past: '%s पहिले',
    m: '१ मिनेट',
    mm: '%d मिनेट',
    h: '१ घण्टा',
    hh: '{x} घण्टा',
  },
});
