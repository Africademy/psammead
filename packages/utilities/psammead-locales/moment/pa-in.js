var moment = require('moment');
require('moment/locale/pa-in');

moment.updateLocale('pa-in', {
  months: ''.split('_'),
  postformat: null,
  relativeTime: {
    past: '%s ਪਹਿਲਾਂ',
  },
});
