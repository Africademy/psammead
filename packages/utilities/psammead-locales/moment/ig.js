import moment from 'moment';
/* eslint-disable */
//! moment.js locale configuration

// ;(function (global, factory) {
//    typeof exports === 'object' && typeof module !== 'undefined'
//        && typeof require === 'function' ? factory(require('../moment')) :
//    typeof define === 'function' && define.amd ? define(['../moment'], factory) :
//    factory(global.moment)
// }(this, (function (moment) { 'use strict';

  // let ig = moment.defineLocale('ig', {
  moment.defineLocale('ig', {
    months: 'Jenụwarị_Febụwarị_Maachị_Eprel_Mee_Juun_Julaị_Ọgọọst_Septemba_Ọktọba_Nọvemba_Disemba'.split(
      '_',
    ),
    monthsShort: 'Jen_Feb_Maa_Epr_Mee_Juu_Jul_Ọgọ_Sep_Ọkt_Nov_Dis'.split('_'),
    weekdays: 'Sọnde_Mọnde_Tuzde_Wenesde_Tọsde_Fraịde_Satọde'.split(
      '_',
    ),
    weekdaysShort: 'Sọn_Mọn_Tuz_We_Tọs_Frai_Sat'.split('_'),
    weekdaysMin: 'Su_Mọ_Tu_We_Tọ_Fr_Sa'.split('_'),
    longDateFormat: {
      LT: 'HH:mm',
      LTS: 'HH:mm:ss',
      L: 'DD/MM/YYYY',
      LL: 'D MMMM YYYY',
      LLL: 'D MMMM YYYY HH:mm',
      LLLL: 'dddd, D MMMM YYYY HH:mm',
    },
    calendar: {
      sameDay: '[Taa na] LT',
      nextDay: '[Echi na] LT',
      nextWeek: 'dddd [na] LT',
      lastDay: '[Unyaahụ na] LT',
      lastWeek: 'dddd [gara aga na] LT',
      sameElse: 'L',
    },
    relativeTime: {
      future: 'na %s',
      past: '%s gara aga',
      s: 'ntabịanya ole na ole',
      ss: 'ntabịanya %d',
      m: 'otu nkeji',
      mm: 'nkeji %d',
      h: 'otu elekere',
      hh: 'elekere %d',
      d: 'otu ụbọchị',
      dd: 'Ụbọchị %d',
      M: 'otu ọnwa',
      MM: 'Ọnwa %d',
      y: 'Otu afọ',
      yy: 'Afọ %d',
    },
    dayOfMonthOrdinalParse: /(Nke )\d{1,2}/,
    ordinal : function (number) {
        return 'Nke ' + number;
    },
    week: {
      dow: 1, // Monday is the first day of the week.
      doy: 4, // The week that contains Jan 4th is the first week of the year.
    },
  });

//   return ig;
// });
