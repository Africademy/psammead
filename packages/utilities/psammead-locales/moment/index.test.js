import moment from 'moment';
import './ig';

const config = [
  {
    format: 'L',
    timestamp: 1520355934123,
  },
];

const locales = [{ locale: 'ig', localeName: 'Igbo' }];

locales.forEach(({ locale, localeName }) => {
  config.forEach(({ format, timestamp }) => {
    const enGB = moment(timestamp)
      .locale('en-gb')
      .format(format);

    it(`should localise ${enGB} to ${localeName}`, () => {
      moment.locale(locale);
      expect(
        moment(timestamp)
          .locale('en-gb')
          .format(format),
      ).toMatchSnapshot();
    });
  });
});
