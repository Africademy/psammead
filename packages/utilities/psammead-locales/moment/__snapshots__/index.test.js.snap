// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should localise to Igbo - '06/03/2018' 1`] = `"06/03/2018"`;

exports[`should localise to Igbo - '6 Mar 2018 17:05' 1`] = `"6 Maa 2018 17:05"`;

exports[`should localise to Igbo - '6 Mar 2018' 1`] = `"6 Maa 2018"`;

exports[`should localise to Igbo - '6 March 2018 17:05' 1`] = `"6 Maachị 2018 17:05"`;

exports[`should localise to Igbo - '6 March 2018' 1`] = `"6 Maachị 2018"`;

exports[`should localise to Igbo - '6/3/2018' 1`] = `"6/3/2018"`;

exports[`should localise to Igbo - '17:05' 1`] = `"17:05"`;

exports[`should localise to Igbo - '17:05:34' 1`] = `"17:05:34"`;

exports[`should localise to Igbo - 'Tue, 6 Mar 2018 17:05' 1`] = `"Tuz, 6 Maa 2018 17:05"`;

exports[`should localise to Igbo - 'Tuesday, 6 March 2018 17:05' 1`] = `"Tuzde, 6 Maachị 2018 17:05"`;
